include/libdnf5/rpm/rpm_signature.hpp

```diff
@@ -81,41 +81,33 @@ class KeyInfo {
-
-    explicit RpmSignature(const libdnf5::BaseWeakPtr & base) : base(base) {}
-    explicit RpmSignature(Base & base) : RpmSignature(base.get_weak_ptr()) {}
-    ~RpmSignature(){};
-
-    /// Check signature of the `package` using public keys stored in rpm database.
-    /// @param package: package to check.
-    /// @return CheckResult::OK - the check passed
-    ///         CheckResult::SKIPPED - the check was skipped
-    ///         CheckResult::FAILED_KEY_MISSING - no corresponding key found in rpmdb
-    ///         CheckResult::FAILED_NOT_TRUSTED - signature is valid but the key is not trusted
-    ///         CheckResult::FAILED_NOT_SIGNED - package is not signed but signature is required
-    ///         CheckResult::FAILED - check failed for another reason
-    CheckResult check_package_signature(Package package) const;
-
-    /// Import public key into rpm database.
-    /// @param key: GPG key to be imported into rpm database.
-    bool import_key(const KeyInfo & key) const;
-
-    /// Check public key presence in rpm database
-    bool key_present(const KeyInfo & key) const;
-
-    /// Download the key file if needed and return a vector of keys contained in it
-    /// @param key_url: URL of the public key to be imported.
-    std::vector<KeyInfo> parse_key_file(const std::string & key_url);
-
-    /// Return string representation of the CheckResult enum
-    static std::string check_result_to_string(CheckResult result);
-
-private:
-    BaseWeakPtr base;
-};
+namespace rpm_signature {
+enum class CheckResult { OK, SKIPPED, FAILED_KEY_MISSING, FAILED_NOT_TRUSTED, FAILED_NOT_SIGNED, FAILED };
+
+/// Check signature of the `package` using public keys stored in rpm database.
+/// @param package: package to check.
+/// @return CheckResult::OK - the check passed
+///         CheckResult::SKIPPED - the check was skipped
+///         CheckResult::FAILED_KEY_MISSING - no corresponding key found in rpmdb
+///         CheckResult::FAILED_NOT_TRUSTED - signature is valid but the key is not trusted
+///         CheckResult::FAILED_NOT_SIGNED - package is not signed but signature is required
+///         CheckResult::FAILED - check failed for another reason
+CheckResult check_package_signature(Base & base, const Package & package);
+
+/// Import public key into rpm database.
+/// @param key: GPG key to be imported into rpm database.
+bool import_key(Base & base, const KeyInfo & key);
+
+/// Check public key presence in rpm database
+bool key_present(Base & base, const KeyInfo & key);
+
+/// Download the key file if needed and return a vector of keys contained in it
+/// @param key_url: URL of the public key to be imported.
+std::vector<KeyInfo> parse_key_file(Base & base, const std::string & key_url);
+
+/// Return string representation of the CheckResult enum
+std::string check_result_to_string(CheckResult result);
+};  // namespace rpm_signature
```

Comment:
I am not sure about this. I'd rather leave the class there and maybe rename it to `RpmSignatureVerifier`. I agree pimpl is not needed and currently changes are not expected for this class, right?

Reply:
Actually this might be changing in future.
`rpm` provides ugly API for checking signatures: https://github.com/rpm-software-management/dnf5/blob/8f46d64fb1a44b307bae5fed7ce5ba233ea0ab0e/libdnf/rpm/rpm_signature.cpp#L107-L143

And there is an `RFE` for something better: https://github.com/rpm-software-management/rpm/issues/2041

Perhaps it would behoove us to add the `pImpl` to be ready for anything.

Reply:
OK makes sense then. So class + pimpl then?